Add Conda environment setup and installation instructions

rodrigo2000m opened the pull request:
Provides an easy, reproducible way to manage dependencies and isolate the project environment, improving setup and development workflow with editable installs.

rodrigo2000m committed:
Add Conda environment setup and installation instructions

petrusen commented:
Thank you, Rodrigo, for your contribution, and apologies for answering so late. 
I agree that having a Conda installation is very helpful. However, I wonder if all the pip packages in the `environment.yml` are strictly necessary. Would it be possible to reduce it to a minimal working list?

Best regards,
Enric

rodrigo2000m commented:
The yml file was automatically generated based on the packages installed following the instructions in the repository. It includes all the packages installed using pip. I considered this a cleaner approach to installation to prevent interference with other programs.
Best regards,
Rodrigo

dgarayr commented:
Hi,
first, thanks @rodrigo2000m for the contribution! I agree that simplifying the setup of conda environments can be a nice feature for several users.

The problem I see in the current `environment.yml` file is that there is a mix-up of packages installed via conda and packages installed via pip. This is likely to lead to conflicts if the environment gets more complex later, and it is a bit against the purposes of isolation and compartimentalization that we need enviroments for. 

I would suggest to regenerate the `environment.yml` file but ensuring that the main dependencies (these listed in the `setup.py`: numpy, matplotlib, networkx, scipy, pandas, scikit-learn, seaborn) are installed via conda, so eventually the only entry under *pip* is POMSimulator itself
```yaml
- pip:
  - pomsimulator==2.0.1
```

If you can make this change, we'll be happy to accept the request.

Thanks again and best regards,
Diego

rodrigo2000m committed:
Refactor environment.yml to include packages directly

Removed pip dependencies from the pip section and added them directly to the environment.

rodrigo2000m commented:
The requested change has been made. Thanks for considering the change.
Best regards,
Rodrigo

petrusen commented:
Thank you, Rodrigo! We’ve added you to the list of further contributors: [CONTRIBUTING.md](https://github.com/petrusen/pomsimulator/blob/main/CONTRIBUTING.md)

Best regards,
Enric
